aten/src/ATen/native/cuda/group_norm_kernel.cu

```diff
@@ -614,21 +726,35 @@ void GroupNormKernelImplInternal(
     // using manual kernel here. Make it using gpu_kernel_multiple_outputs once
     // the issue fixed.
     const int64_t B = (N * C + kCUDANumThreads - 1) / kCUDANumThreads;
+
     ComputeFusedParamsCUDAKernel<T><<<B, kCUDANumThreads, 0, cuda_stream>>>(
         N, C, G, mean_data, rstd_data, gamma_data, beta_data, a_data, b_data);
     C10_CUDA_KERNEL_LAUNCH_CHECK();
 
-    auto iter = TensorIteratorConfig()
-                    .check_all_same_dtype(std::is_same<T, T_ACC>::value)
-                    .resize_outputs(false)
-                    .add_owned_output(Y.view({N * C, HxW}))
-                    .add_owned_const_input(X.view({N * C, HxW}))
-                    .add_owned_input(a.view({N * C, 1}))
-                    .add_owned_input(b.view({N * C, 1}))
-                    .build();
-    gpu_kernel(iter, [] GPU_LAMBDA(T x, T_ACC a, T_ACC b) -> T {
-      return a * static_cast<T_ACC>(x) + b;
-    });
+    switch(x_format) {
+      case MemoryFormat::Contiguous: {
+          TensorIterator iter = TensorIteratorConfig()
+                      .check_all_same_dtype(std::is_same<T, T_ACC>::value)
+                      .resize_outputs(false)
+                      .add_owned_output(Y.view({N * C, HxW}))
+                      .add_owned_const_input(X.view({N * C, HxW}))
+                      .add_owned_input(a.view({N * C, 1}))
+                      .add_owned_input(b.view({N * C, 1}))
+                      .build();
+          gpu_kernel(iter, [] GPU_LAMBDA(T x, T_ACC a, T_ACC b) -> T {
+            return a * static_cast<T_ACC>(x) + b;
+          });
+          break;
+      }
+      case MemoryFormat::ChannelsLast: {
+            ApplyScaleBiasNHWCKernel<T><<<N * G, num_threads, 0, cuda_stream>>>(X_data, Y_data, N, height, width, C, D*HxW, a_data, b_data);
+            C10_CUDA_KERNEL_LAUNCH_CHECK();
+            break;
```

Comment:
I do not know how to do this with a tensor iterator. IF someone could show me how to that would be great.